solutions/CSES Subarray Sums II.mdx

```diff
@@ -0,0 +1,173 @@
+---
+id: cses-1661
+source: CSES
+title: Subarray Sums II
+author: Qi Wang
+---
+
+# Problem
+
+We are asked to find the number of subarrays that sum up to $x$ given the size of the array and its elements.
+
+<Spoiler title="Solution">
+
+We can have a map that keeps track of the prefix sums, we can check whether there are subarrays with prefix sum equal to $\texttt{prefixSum} - x$. This will ensure that we can remove/add the sum of the subarray that has $\texttt{prefixSum} - x$ from/to our current subarray to build our answer.
```

Comment:
> subarrays with prefix sum equal to

replace this with "prefixes with sum equal to"

> This will ensure that we can remove/add the sum of the subarray that has $\texttt{prefixSum} - x$ from/to our current subarray to build our answer.

remove a prefix from our current prefix to build a subarray with sum $x$